robot_modeling/DQ_Kinematics.m

```diff
@@ -367,6 +369,29 @@ function set_base_frame(obj, base_frame)
             J = 2*vec4(robot_line - workspace_line)'*Jl;
             
         end
+
+
+        function residual = line_to_line_angle_residual(robot_line, ...
+                 workspace_line, workspace_line_derivative) 
```

Comment:
Since only the line *direction* is used for those calculations (i.e., only the primary part of the line), I suggest the following modification:

`robot_line` -> `robot_line_direction` (and the same for the workspace line and its derivative).